cpp/CMakeLists.txt

```
@@ -89,15 +89,38 @@ if("${GPU_ARCHS}" STREQUAL "")
 endif()
 
 if("${GPU_ARCHS}" STREQUAL "ALL")
-  set(GPU_ARCHS "60")
-  if((CUDA_VERSION_MAJOR EQUAL 9) OR (CUDA_VERSION_MAJOR GREATER 9))
-    set(GPU_ARCHS "${GPU_ARCHS};70")
-  endif()
-  if((CUDA_VERSION_MAJOR EQUAL 10) OR (CUDA_VERSION_MAJOR GREATER 10))
-    set(GPU_ARCHS "${GPU_ARCHS};75")
+  
+  # Check for embedded vs workstation architectures
+  if(CMAKE_SYSTEM_PROCESSOR MATCHES "aarch64")
+    # This is being built for Linux4Tegra or SBSA ARM64
+    message(STATUS "Detected aarch64 cpu architecture: using L4T and SBSA compatible GPU_ARCHS")
+    set(GPU_ARCHS "62")
+    if((CUDA_VERSION_MAJOR EQUAL 9) OR (CUDA_VERSION_MAJOR GREATER 9))
+      set(GPU_ARCHS "${GPU_ARCHS};72")
+    endif()
+    if((CUDA_VERSION_MAJOR EQUAL 11) OR (CUDA_VERSION_MAJOR GREATER 11))
```

These should all be simplified using `GREATER_EQUAL`
```suggestion
    if(CUDA_VERSION_MAJOR GREATER_EQUAL 9)
      set(GPU_ARCHS "${GPU_ARCHS};72")
    endif()
    if(CUDA_VERSION_MAJOR GREATER_EQUAL 11)
```